WordPress/src/main/java/org/xmlrpc/android/XMLRPCClient.java

```diff
@@ -319,8 +319,17 @@ public static Object parseXMLRPCResponse(InputStream is, HttpEntity entity)
             // no parser.require() here since its called in XMLRPCSerializer.deserialize() below
             // deserialize fault result
             Map<String, Object> map = (Map<String, Object>) XMLRPCSerializer.deserialize(pullParser);
-            String faultString = (String) map.get(TAG_FAULT_STRING);
-            int faultCode = (Integer) map.get(TAG_FAULT_CODE);
+            //Check that required tags are in the response
+            if (!map.containsKey(TAG_FAULT_STRING) || !map.containsKey(TAG_FAULT_CODE)) {
+                throw new XMLRPCException("Bad XMLRPC Fault response received - neither <faultCode> nor <faultString>");
+            }
+            String faultString = String.valueOf(map.get(TAG_FAULT_STRING));
+            int faultCode;
+            try {
+                faultCode = (int) map.get(TAG_FAULT_CODE);
+            } catch (NumberFormatException | ClassCastException e) {
+                throw new XMLRPCException("Bad XMLRPC Fault response received - <faultCode> value is not an integer");
+            }
             consumeHttpEntity(entity);
```

Comment:
Right. I changed the code by moving `consumeHttpEntity` in the correct place. It was not even necessary to consume the entity in this case, since the whole parsing fails and the connection is closed.
XML-RPC response is just one, we're not using XML-RPC multicalls in the app.